resources/js/components/MainLayout/LogoutButton/LogoutButton.js

```diff
@@ -39,4 +40,4 @@ function LogoutButton({history}) {
     )
 }
 
-export default withRouter(LogoutButton)
+export default useNavigate(LogoutButton)
```

Comment:
The useNavigate() hook is introduced in the React Router v6 to replace the useHistory() hook.  For reference please check this -> https://www.geeksforgeeks.org/reactjs-usenavigate-hook/ and  `withRouter` does not work in v6. I am new full-stack developer but new to React. So I followed your article to paly around. I got some errors and warning, and I replace those with latest one. All errors solutions are found on google.

Reply:
Yes, the article is correct, from React Router v6 you should use useNavigate instead of withRouter, but you're using the wrong way, useNavigate is a hook not a higher-level component like the withRouter, and that's why you shouldn't use this:
`useNavigate(LogouthButton)`
, as you can see in the article too, you should use as a variable:
`let navigate = useNavigate()`

Reply:
Thanks, got it.